mono/sgen/sgen-los.c

```diff
@@ -158,7 +158,7 @@ los_consistency_check (void)
 
 	FOREACH_LOS_OBJECT_NO_LOCK (obj) {
 		mword obj_size = sgen_los_object_size (obj);
-		char *end = obj->data + obj_size;
+		char *end = (char*)obj->data + obj_size;
```

Comment:
Not sure if this is really correct since data is not technically a pointer. Wouldn't be surprised if it doesn't work on all compilers out there. I'd say it is better to use `(char*)&obj->data [0]`.

Reply:
Hmm, seems true, as it is a zero-length array on most platforms. I'll check that and push a new commit later.

Reply:
Ok, changed that. It should be fine now, ARM64 and Windows x86_64 still accepts it. I'm not able to check other platforms.